### ISSUE TITLE
Thumbnail letterbox issues

### ISSUE THREAD

**Issue by Repetria** [axcore/tartube#661]
Thanks for taking the time to submit a bug report! Please include the following information:

### What operating system are you using?
windows 11

### What version of Tartube are you using?
v2.5.040
![Untitled design](https://github.com/user-attachments/assets/691b86fb-8a4a-44e8-a717-92dbf42899ea)

Hi, Like an image, a letterbox is created on the downloaded video thumbnail. 

So it would be nice to have the ability to re-download just the thumbnails.

Sometimes the issue is printed in the errors/warnings tab, sometimes not, so I would like to be able to re-download the thumbnail of the entire video.

If this is a feature that already exists or something I can try in the current version, please let me know, thanks.

**Comment by axcore** [comment]
I couldn't find a simple way to do that, so I added some new code.

You can now "check" videos that have already been downloaded.  

For example, in the Videos tab, right-click a video and select **Re-check video**. This will delete thumbnails and other metadata files, before downloading them again. (The actual video is not deleted.)

If you right-click a video and you see **Check video** instead of **Re-check video**, then no files are deleted. But while checking the video, any new files will be downloaded as normal. (Previously, nothing happened when you tried to do this.)

The new code is uploaded to github/pypi only. You can install the updated code now [as described here](https://github.com/axcore/tartube?tab=readme-ov-file#72-update-tartube-on-ms-windows), or you can wait for the full release in a few days.

**Comment by Repetria** [comment]
thanks! it was so helpful

2024년 10월 29일 (화) 오전 1:33, A S Lewis ***@***.***>님이 작성:

**Comment by Repetria** [comment]
Hi, I'm leaving a comment because I have a question. 
How does re-check video work? 

I don't know how to read the code, but if the logic is like this 
Delete thumbnail, metadata -> download thumbnail from YouTube again. 

I think this would be better :
Check if the video exists on YouTube (if not, ignore the next step) -> delete thumbnail, metadata -> download thumbnail from YouTube again. 

If you don't do this, you won't be able to download thumbnails for deleted/missing channel videos on youtube. 
I apologize if I sounded rude. Thank you for always doing your best.

**Comment by axcore** [comment]
Yes, "re-download video" deletes all files, including the thumbnail, "re-check video" deletes all the files except the video itself.

I will think about a backup, in case the video is no longer available.

**Comment by axcore** [comment]
v2.5.100 adds a new feature. When you re-check or re-download a video, its files are copied to a temporary folder, instead of being deleted.

If the re-check/re-download fails, then the original files are recovered.

The process is not foolproof, but should work nearly all of the time.
